apps/dashboard/app/lib/smart_attributes/attributes/auto_modules.rb

```diff
@@ -31,31 +31,52 @@ def label(*)
       end
 
       def select_choices(*)
-        versions = HpcModule.all_versions(@hpc_module).map do |mod|
-          data_opts = Configuration.job_clusters.map do |cluster|
-            { "data-option-for-cluster-#{cluster.id}": false } unless mod.on_cluster?(cluster.id)
-          end.compact
-
-          [mod.version, mod.to_s].concat(data_opts)
-        end
-
-        if show_default?
-          versions.prepend(['default', @hpc_module])
-        else
-          versions
-        end
+        versions = hpc_versions
+        versions = versions_with_default(versions) if show_default?
+        versions = filtered_versions(versions) if filter_versions?
+        versions
       end
 
       def show_default?
         default = @opts[:default]
         default.nil? ? true : default != false
       end
 
+      def filter
+        @filter ||= @opts[:filter]
```

Comment:
Do we want an array of filters or just instruct folks to use `|` blocks?

Reply:
I am unsure what you mean by `|` blocks - what would that look like?

Reply:
Sorry - OR `|` in regular expressions. Like supplying a regex to filter 2 things: `*the_thing|*the_other_thing`

Reply:
Gotcha. I think I'll rewrite it to accept an array - I think it'll encourage simpler use.